src/QMCDrivers/CloneManager.cpp

```diff
@@ -50,13 +50,30 @@ std::vector<std::vector<MCWalkerConfiguration*> > CloneManager::WPoolClones;
 std::vector<std::vector<TrialWaveFunction*> > CloneManager::PsiPoolClones;
 std::vector<std::vector<QMCHamiltonian*> > CloneManager::HPoolClones;
 
+
+// Clear the static clones.  For now only clear wClones so makeClones will work.
+// Used in unit tests.
```

Comment:
Please emphasize more that it is not for general use and should be used only by unit tests. Please also add the same comment in header file.

Reply:
In general, I don't think the clear method is safe since it erases the vector of pointers without freeing the pointers.

Reply:
I'll add more comments and change the name to 'clear_for_unit_tests'

Not freeing the pointers is safe - it will leak memory, but that's okay for now.   The alternative is to free all the pointers, and then we have to be very careful put only dynamically allocated objects in there (that is, it will crash the unit tests as currently written).   Ultimately, the correct solution is to use some form of smart pointer.

Reply:
Renaming the function is a very good solution. As long as it is sufficient for unit tests, no need to spend time investigating how to make it fully safe.